Source/EasyNetQ/EasyNetQBuilderExtensions.cs

```diff
@@ -0,0 +1,154 @@
+using EasyNetQ.ChannelDispatcher;
+using EasyNetQ.Consumer;
+using EasyNetQ.Interception;
+using EasyNetQ.MessageVersioning;
+using EasyNetQ.MultipleExchange;
+using EasyNetQ.Persistent;
+using EasyNetQ.Producer;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace EasyNetQ;
+
+public static class EasyNetQBuilderExtensions
+{
+    public static IEasyNetQBuilder EnableMultiChannelClientCommandDispatcher(
+        this IEasyNetQBuilder builder, int channelsCount
+    )
+    {
+        builder.Services.AddSingleton<IPersistentChannelDispatcher>(
+            x => new MultiPersistentChannelDispatcher(
+                channelsCount,
+                x.GetRequiredService<IProducerConnection>(),
+                x.GetRequiredService<IConsumerConnection>(),
+                x.GetRequiredService<IPersistentChannelFactory>()
+            )
+        );
+        return builder;
+    }
+
+    public static IEasyNetQBuilder EnableLegacyTypeNaming(this IEasyNetQBuilder builder)
+    {
+        builder.Services.AddSingleton<ITypeNameSerializer, LegacyTypeNameSerializer>();
+        return builder;
+    }
+
+    public static IEasyNetQBuilder EnableLegacyRpcConventions(this IEasyNetQBuilder builder)
+    {
+        builder.Services.AddSingleton<IConventions, LegacyRpcConventions>();
+        return builder;
+    }
+
+    public static IEasyNetQBuilder EnableLegacyConventions(this IEasyNetQBuilder builder)
+    {
+        return builder
+            .EnableLegacyTypeNaming()
+            .EnableLegacyRpcConventions();
+    }
+
+    public static IEasyNetQBuilder EnableDelayedExchangeScheduler(this IEasyNetQBuilder builder)
+    {
+        builder.Services.AddSingleton<IScheduler, DelayedExchangeScheduler>();
+        return builder;
+    }
+
+    public static IEasyNetQBuilder EnableAdvancedMessagePolymorphism(this IEasyNetQBuilder builder)
+    {
+        builder.Services.AddSingleton<IExchangeDeclareStrategy, MultipleExchangeDeclareStrategy>();
+        return builder;
+    }
+
+    public static IEasyNetQBuilder EnableMessageVersioning(this IEasyNetQBuilder builder)
+    {
+        builder.Services
+            .AddSingleton<IExchangeDeclareStrategy, VersionedExchangeDeclareStrategy>()
+            .AddSingleton<IMessageSerializationStrategy, VersionedMessageSerializationStrategy>();
+        return builder;
+    }
+
+    public static IEasyNetQBuilder EnableAlwaysAckConsumerErrorStrategy(this IEasyNetQBuilder builder)
+    {
+        builder.Services.AddSingleton<IConsumeErrorStrategy>(SimpleConsumeErrorStrategy.Ack);
```

Comment:
Also, one more thing.

Our MS DI adapter used Replace mechanics instead of Add. Should we do the same here?

Reply:
I guess it should be replace here because it's called after the default services are registered right?